kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,275 @@
+      return result;
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   */
+  public Set<Topic> getConsumerGroupTopicInformation(String groupId,
+                                                         Option<Integer> offsetOpt,
+                                                         Option<Integer> countOpt)
+          throws Exception {
+    if (Option.apply(adminClientWrapper).nonEmpty()) {
+      final Collection<MemberDescription> summariesOpt =
+              adminClientWrapper.describeConsumerGroups(Collections.singleton(groupId))
+                      .get(groupId).members();
+      if (summariesOpt.size() > 0) {
+        final Set<Topic> result = new HashSet<>();
+        for (MemberDescription eachSummary : summariesOpt) {
+          for (TopicPartition topicPartition : eachSummary.assignment().topicPartitions()) {
+            result.add(new Topic(topicPartition.topic(), null, null));
+          }
+        }
+        log.debug("Get topic list {}", result);
+        final boolean needPartOfData = offsetOpt.nonEmpty()
+                && countOpt.nonEmpty()
+                && countOpt.get() > 0;
+        if (!needPartOfData) {
+          return result;
+        } else {
+          final scala.collection.Set<Topic> slice =
+                  JavaConverters.asScalaSetConverter(result).asScala()
+                          .slice(offsetOpt.get(), offsetOpt.get() + countOpt.get()).toSet();
+          return JavaConverters.setAsJavaSetConverter(slice).asJava();
+        }
+      } else {
+        return Collections.emptySet();
+      }
+    } else {
+      return Collections.emptySet();
+    }
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   *     (all consumed topics with all partition offset information)
+   */
+  public ConsumerGroupSubscription getConsumerGroupInformation(String groupId) throws Exception {
+    return getConsumerGroupInformation(groupId, Option.<String>empty(),
+            Option.<Integer>empty(), Option.<Integer>empty());
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId   - group name
+   * @param topic     - topic name for filter - default empty topic name
+   * @param offsetOpt - offset for TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @param countOpt  - count of elements TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @return description of consumer group
+   */
+  public ConsumerGroupSubscription getConsumerGroupInformation(
+          String groupId,
+          Option<String> topic,
+          Option<Integer> offsetOpt,
+          Option<Integer> countOpt) throws Exception {
+    if (Option.apply(adminClientWrapper).nonEmpty()) {
```

Comment:
Make sure adminClientWrapper is non-null during construction, then this check won't be needed anymore.